akd_client/src/verify.rs

```diff
@@ -135,6 +136,23 @@ fn verify_vrf(
     vrf_pk.verify_label(uname, stale, version, pi, label)
 }
 
+/// Verifies a serialized JSON lookup proof after deserializing it first.
+pub fn serialized_lookup_verify(
+    _vrf_public_key: &[u8],
+    root_hash: Digest,
+    _akd_key: AkdLabel,
+    serialized_json_proof: &str,
```

Comment:
So this is going to be a non-backwards compatible JSON string, since we're using `serde` which doesn't guarantee backwards compat right? If we are expecting this data to come over the wire, we should really be using something like protobuf unfortunately... :( Even the same structure, depending on json serializer, can have breakable differences in the serialized payload...

Reply:
If the JSON serializer/deserializer are incompatible for the same structure, we would expect a major version difference between them since such a change would be breaking. I don't see this issue being applicable to `serde_json` since it is quite stable at this point and there hasn't been even a minor version bump in the last 5 years.

Overall, as long as we follow the general backward compatibility rules (e.g., never remove a field) I don't see much difference between `serde_json` and `protobuf` from backward compatibility perspective -- except `protobuf` requiring more manual work to write the structs, generate them by hand and re-parsing the primitive parts individually and mapping them to the Rust types (which is what we are trying to avoid). Is it possible to just do `let proof: LookupProof = protobuf:from_str(&serialized_proof)`? 

I also looked into using Thrift but `fbthrift` crate has not been updated in quite some time.